internal/fsutil/create.go

```diff
@@ -51,8 +57,13 @@ func Create(options *CreateOptions) (*Entry, error) {
 
 	switch o.Mode & fs.ModeType {
 	case 0:
-		err = createFile(o)
-		hash = hex.EncodeToString(rp.h.Sum(nil))
+		if o.Link != "" {
+			err = createHardLink(o)
+			hardLink = true
```

Comment:
Why do we need the boolean flag if we have the mode already?  Looks like we've done without it for the Options, so the Entry could be the same.

Reply:
There is an edge case where we are creating a hard link to a symlink. In that case, we are asking `fsutil.Create` to create an entry with `o.Link = "symlink_location"` and `o.Mode & fs.ModeType = 0` (meaning no dir or symlink). However, the final entry will have the mode of the file that was created, if it points to `symlink_location` then the mode will have the symlink bit set. The problem then is that we are not able to distinguish hard link to symlink from a plain symlink because `Mode` and `Link` match both.